walnuts_cpp/test_stan.cpp

```diff
@@ -0,0 +1,188 @@
+#include <iostream>
+#include <memory>
+#include <random>
+#include <cmath>
+#include <chrono>
+
+#include <Eigen/Dense>
+#include <bridgestan.h>
+
+#include "nuts.hpp"
+
+// consider using something like https://github.com/martin-olivier/dylib/
+#ifdef _WIN32
+// hacky way to get dlopen and friends on Windows
+
+#include <errhandlingapi.h>
+#define dlopen(lib, flags) LoadLibraryA(lib)
+#define dlsym(handle, sym) (void*)GetProcAddress(handle, sym)
+#define dlclose(handle) FreeLibrary(handle)
+
+char* dlerror() {
+  DWORD err = GetLastError();
+  int length = snprintf(NULL, 0, "%d", err);
+  char* str = malloc(length + 1);
+  snprintf(str, length + 1, "%d", err);
+  return str;
+}
+#else
+#include <dlfcn.h>
+#endif
+
+struct dlclose_deleter {
+  void operator()(void* handle) const {
+    if (handle) {
+      dlclose(handle);
+    }
+  }
+};
+
+auto dlopen_safe(const char *path) {
+  auto handle = dlopen(path, RTLD_NOW);
+  if (!handle) {
+    throw std::runtime_error(std::string("Error loading library '") + path +
+                             "': " + dlerror());
+  }
+  return std::unique_ptr<void, dlclose_deleter>(handle);
+}
+
+template <typename U, typename T>
+auto dlsym_cast(U &library, T &&, const char *name) {
+  auto sym = dlsym(library.get(), name);
+  if (!sym) {
+    throw std::runtime_error(std::string("Error loading symbol '") + name +
+                             "': " + dlerror());
+  }
+  return reinterpret_cast<T>(sym);
+}
+
+double total_time = 0.0;
+int count = 0;
+
+template<typename T>
+void nop_deleter(T*){}
```

Comment:
How about `no_op_deleter`?   The double use of the `o` is bugging me :-).

Reply:
Fine by me. I think my brain latched on to “nop” as that’s the name for the no-op instruction in most of the common assembly formats